jms/deployment/src/main/java/io/quarkus/artemis/jms/deployment/ArtemisJmsProcessor.java

```diff
@@ -42,30 +45,29 @@ HealthBuildItem health(ArtemisBuildTimeConfig buildConfig) {
     @Record(ExecutionTime.RUNTIME_INIT)
     @BuildStep
     ArtemisJmsConfiguredBuildItem configure(ArtemisJmsRecorder recorder, ArtemisBuildTimeConfig config,
+            Optional<ArtemisJmsWrapperBuildItem> wrapper,
+            Capabilities capabilities,
             BuildProducer<SyntheticBeanBuildItem> syntheticBeanProducer) {
-        SyntheticBeanBuildItem connectionFactory = SyntheticBeanBuildItem
-                .configure(ActiveMQJMSConnectionFactory.class)
-                .addType(ConnectionFactory.class)
-                .supplier(recorder.getConnectionFactorySupplier())
+
+        SyntheticBeanBuildItem.ExtendedBeanConfigurator configurator;
+        if (config.xaEnabled) {
+            configurator = SyntheticBeanBuildItem.configure(ActiveMQConnectionFactory.class);
```

Comment:
Since `ActiveMQConnectionFactory` implements both `ConnectionFactory` and `XAConnectionFactory`, even with `quarkus.artemis.xa.enabled=true` we still need to export `ConnectionFactory` which is used by `ConnectionFactoryHealthCheck`  for doing the health check.

Reply:
Ah, I see the `addType` is after the if else. So line 60 `configurator.addType(ConnectionFactory.class)` could be placed inside the `if`.

Reply:
yeah, I agree and that makes sense. I will add some comments inside `if` statment.